### ISSUE TITLE
BASH_VERSION environment variable is set when not using bash shell

### ISSUE THREAD

**Issue by taylorthurlow** [Microsoft/vscode#66798]
- VSCode Version: 1.30.2
- OS Version: Ubuntu 18.04.1

Steps to Reproduce:

1. Use `zsh` as the default shell
2. Open an integrated terminal
3. Both `echo $BASH_VERSION` and `echo $ZSH_VERSION` return their respective versions, even though bash is not being used. This is breaking a git info plugin I use with `oh-my-zsh` - it thinks it's running bash because the var is set.

Does this issue occur when all extensions are disabled?: Yes

----

Is this intentional? Can/should it be fixed?

**Comment by Tyriar** [comment]
Likely related to CLI environment setup

**Comment by joaomoreno** [comment]
> Likely related to CLI environment setup

@Tyriar can you elaborate?

**Comment by joaomoreno** [comment]
@taylorthurlow Can you record a screenrecording of what you mean?

**Comment by taylorthurlow** [comment]
@joaomoreno No need for a screen recording, a couple of images should explain well:

An integrated terminal in VScode:
![2019-01-23-111608_381x147_scrot](https://user-images.githubusercontent.com/761640/51631040-7ed72200-1f00-11e9-820e-9202f06e8812.png)
A standalone, separate terminal:
![2019-01-23-111618_479x189_scrot](https://user-images.githubusercontent.com/761640/51631054-88608a00-1f00-11e9-93f1-e085287e0ab9.png)

The fact that `$BASH_VERSION` is set in the integrated terminal seems either highly inconvenient or just wrong, given that I'm using `zsh`. Currently it's breaking an `oh-my-zsh` plugin which uses `$BASH_VERSION` in a conditional.

**Comment by Tyriar** [comment]
@joaomoreno if $SHELL is set to bash when VS Code is launched I would expect this to happen because of:

https://github.com/Microsoft/vscode/blob/c45d91d9c147f05066c41db13eb28b3ad3cacd44/src/vs/code/node/shellEnv.ts#L24

@taylorthurlow can you run VS Code from the dock/dash then open devtools (Help > Toggle Developer Tools), open the console tab and run process.env and look for SHELL, BASH_VERSION and ZSH_VERSION? My suspicion is that $SHELL is set to bash as it's not being set to zsh on login in which case this is working as designed.

**Comment by taylorthurlow** [comment]
Here's the relevant entries of `process.env` after opening from the equivalent of a dock (I assume basically we just want to open from something which loads entries from *.desktop files, in my case I'm using `dmenu`):

``` json
{
  "BASH_VERSION": "4.4.19(1)-release",
  "SHELL": "/usr/bin/zsh",
}
```
I didn't find `ZSH_VERSION` in the list. Here's a gist with the entire response:
https://gist.github.com/taylorthurlow/be2095368abe43c174bb88d19ab4c5c6

**Comment by Tyriar** [comment]
@taylorthurlow so I'm thinking my guess is correct. I think you're probably setting $SHELL in your `~/.bashrc` or something to zsh, otherwise this:

https://github.com/Microsoft/vscode/blob/c45d91d9c147f05066c41db13eb28b3ad3cacd44/src/vs/code/node/shellEnv.ts#L24

would have run zsh (`~/.zshrc`) and not set `$BASH_VERSION`.

**Comment by joaomoreno** [comment]
If you open the DevTools and run `getLazyEnv().then(r => console.log(r))` does `BASH_VERSION` appear in that object?

**Comment by taylorthurlow** [comment]
Yes - both `BASH_VERSION` and `SHELL` are set with the same values as the previous look at `process.env`.

**Comment by joaomoreno** [comment]
For some reason your shell seems to be setting this... In a regular shell, does the `$SHELL -ilc 'node -p "process.env"'` command include `BASH_VERSION`?
